src/types.rs

```diff
@@ -577,38 +617,46 @@ impl Term {
             }
         }
     }
-    /// Every [`Atom`] used in the term.
+    /// Returns every [`Atom`] used in the term.
     ///
     /// [`Atom`]: enum.Atom.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Atom};
+    /// let mut sig = Signature::default();
+    /// let var = sig.new_var(Some("Z".to_string()));
+    /// let op = sig.new_op(0, Some("A".to_string()));
+    /// let args = vec![];
+    ///
+    /// let var_term = Term::Variable(var);
+    /// let op_term = Term::Application{op, args};
+    ///
+    /// assert_eq!(var_term.atoms(), vec![Atom::Variable(var)]);
+    /// assert_eq!(op_term.atoms(), vec![Atom::Operator(op)]);
+    /// ```
```

Comment:
As with `Term::display`, how about constructing a more complex term (e.g. `A(B z_))`) and showing the result of these functions there? That would be more informative.